cpp/oneapi/dal/algo/linear_regression/backend/gpu/train_kernel_norm_eq_dpc.cpp

```diff
@@ -37,6 +37,29 @@ using dal::backend::context_gpu;
 namespace be = dal::backend;
 namespace pr = be::primitives;
 
+template <typename Float>
+sycl::event add_ridge_penalty(sycl::queue& q,
```

Comment:
you can add misc.hpp file to avoid code duplication between online and batch(not critical)

Reply:
implementations are slightly different in the online and batch modes, decided to keep them different for now, might change in the future